### PR TITLE
Updating junit output formatter

### PR DESCRIPTION
**Pull request by sushant40**
- Using relative file name. This makes it easier to consume in CI/CD where the checkout directory may change.
- Changing class name to something more meaningful (Swift Lint)
- Minor formatting cleanup to generated file... added tabs and new lines to ensure tags are 'pretty' formatted.

### CODE REVIEW

File: Source/SwiftLintFramework/Reporters/JUnitReporter.swift [hunk 1]
```diff
@@ -10,17 +10,17 @@ public struct JUnitReporter: Reporter {
     }
 
     public static func generateReport(_ violations: [StyleViolation]) -> String {
-        return "<?xml version=\"1.0\" encoding=\"utf-8\"?>\n<testsuites><testsuite>" +
+        return "<?xml version=\"1.0\" encoding=\"utf-8\"?>\n<testsuites><testsuite>\n" +
             violations.map({ violation -> String in
-                let fileName = (violation.location.file ?? "<nopath>").escapedForXML()
+                let fileName = (violation.location.relativeFile ?? "<nopath>").escapedForXML()
```
Comment (marcelofabri) [inline Source/SwiftLintFramework/Reporters/JUnitReporter.swift:15]: can you update the tests so `file != relativePath`?

File: Source/SwiftLintFramework/Reporters/JUnitReporter.swift [hunk 2]
```diff
@@ -10,17 +10,17 @@ public struct JUnitReporter: Reporter {
...
                 let severity = violation.severity.rawValue + ":\n"
                 let message = severity + "Line:" + String(violation.location.line ?? 0) + " "
                 let reason = violation.reason.escapedForXML()
                 return [
-                    "\n\t<testcase classname='Formatting Test' name='\(fileName)\'>\n",
-                    "<failure message='\(reason)\'>" + message + "</failure>",
-                    "\t</testcase>"
+                    "\t<testcase classname='Swift Lint' name='\(fileName)\'>\n",
```
Comment (marcelofabri) [inline Source/SwiftLintFramework/Reporters/JUnitReporter.swift:20]: `SwiftLint` should be spelled as one word

**Comment by marcelofabri** [comment]
Can you also add an entry to the CHANGELOG?

File: Source/SwiftLintFramework/Reporters/JUnitReporter.swift (hunk 2 above)
Comment (sushant40) [reply Source/SwiftLintFramework/Reporters/JUnitReporter.swift:20]: Sure

File: Source/SwiftLintFramework/Reporters/JUnitReporter.swift (hunk 1 above)
Comment (sushant40) [reply Source/SwiftLintFramework/Reporters/JUnitReporter.swift:15]: Not sure I understand what you mean.

Comment (sushant40) [reply Source/SwiftLintFramework/Reporters/JUnitReporter.swift:15]: ... what test file should i be updating?

Comment (marcelofabri) [reply Source/SwiftLintFramework/Reporters/JUnitReporter.swift:15]: check `ReporterTests.swift`

**Pull request closed without merging**